Objects/unicodeobject.c

```diff
@@ -727,15 +778,16 @@ PyUnicode_FromFormatV(const char *format, va_list vargs)
     const char* f;
     Py_UNICODE *s;
     PyObject *string;
-    /* used by sprintf */
-    char buffer[21];
     /* use abuffer instead of buffer, if we need more space
      * (which can happen if there's a format specifier with width). */
     char *abuffer = NULL;
     char *realbuffer;
     Py_ssize_t abuffersize = 0;
-    char fmt[60]; /* should be enough for %0width.precisionld */
     const char *copy;
+    /* used by sprintf */
+    char buffer[64];
+    char fmt[16]; /* should be enough for %0width.precisionld */
```

Comment:
Why are you making this array smaller?

Reply:
Because it doesn't need to be greater, and 16 is already arguably way too big. Look at the comment: This leaves at least 10 bytes for width and precision (I'm counting a second 'l'), which is more than will be ever required I think (I've never seen such parameters set to numbers needing more than 2 digits; even for 3 there is plenty of room). The added benefit of this is to avoid passing too big numbers to `snprintf`, avoiding potential bugs below.

I think the original author(s) mixed up `buffer` (which was arguably too small) and `fmt` (which was way too big).

And it's best that these fields are 8 bytes aligned, at the very least. If you want to change them again, please avoid using silly values (21 or 60) as the original code was doing.

Reply:
OK.

Don't worry about alignment. The C compiler will pad the stack as needed: https://stackoverflow.com/questions/1061818/stack-allocation-padding-and-alignment

Reply:
Yes, wasn't clear, alignment is done by the compiler indeed, but that's why allocations should be by multiples of 4 or 8: Other size of allocations are wasting space.